cmd/orderimports/orderimports.go

```diff
@@ -0,0 +1,226 @@
+				} else {
+					// external repositories are fourth
+					externalImports = append(externalImports, importPath)
+				}
+
+			}
+			if len(originalImports) == 0 {
+				continue
+			}
+			sort.Strings(stdlibImports)
+			sort.Strings(localImports)
+			sort.Strings(k8sImports)
+			sort.Strings(externalImports)
+
+			orderImports := []string{}
+			orderImports = append(orderImports, stdlibImports...)
+			orderImports = append(orderImports, localImports...)
+			orderImports = append(orderImports, k8sImports...)
+			orderImports = append(orderImports, externalImports...)
+			if diff := cmp.Diff(originalImports, orderImports); diff != "" {
+				a.failed = true
+				fmt.Printf("%s (-got +want):\n%s", pathToFile, diff)
+			}
+		}
+	}
+}
+
+func (a *analyzer) filterFiles(fs map[string]*ast.File) []*ast.File {
+	var files []*ast.File
+	for fileName, f := range fs {
+		if a.ignore.MatchString(fileName) {
+			continue
+		}
+		files = append(files, f)
+	}
+	return files
+}
+
+type collector struct {
+	dirs  []string
+	regex *regexp.Regexp
+}
+
+// handlePath walks the filesystem recursively, collecting directories,
+// ignoring some unneeded directories (hidden/vendored) that are handled
+// specially later.
+func (c *collector) handlePath(path string, info os.FileInfo, err error) error {
+	if err != nil {
+		return err
+	}
+	if info.IsDir() {
+		// Ignore hidden directories (.git, .cache, etc)
+		if len(path) > 1 && path[0] == '.' ||
+			// Staging code is symlinked from vendor/k8s.io, and uses import
+			// paths as if it were inside of vendor/. It fails typechecking
+			// inside of staging/, but works when typechecked as part of vendor/.
+			path == "staging" ||
+			// OS-specific vendor code tends to be imported by OS-specific
+			// packages. We recursively typecheck imported vendored packages for
+			// each OS, but don't typecheck everything for every OS.
+			path == "vendor" ||
+			path == "_output" ||
+			// This is a weird one. /testdata/ is *mostly* ignored by Go,
+			// and this translates to kubernetes/vendor not working.
+			// edit/record.go doesn't compile without gopkg.in/yaml.v2
+			// in $GOSRC/$GOROOT (both typecheck and the shell script).
+			path == "pkg/kubectl/cmd/testdata/edit" {
+			return filepath.SkipDir
+		}
+		if c.regex.MatchString(path) {
+			c.dirs = append(c.dirs, path)
+		}
+	}
+	return nil
+}
+
+func main() {
+	flag.Parse()
+	args := flag.Args()
+
+	if len(args) == 0 {
+		args = append(args, ".")
+	}
+
+	regex, err := regexp.Compile(*regex)
+	if err != nil {
+		log.Fatalf("Error compiling regex: %v", err)
+	}
+	c := collector{regex: regex}
+	for _, arg := range args {
+		err := filepath.Walk(arg, c.handlePath)
+		if err != nil {
+			log.Fatalf("Error walking: %v", err)
+		}
+	}
+	sort.Strings(c.dirs)
+
+	if isTerminal {
+		logPrefix = "\r" // clear status bar when printing
+	}
+	fmt.Println("checking-imports-order: ")
+
+	a := newAnalyzer()
+	for _, dir := range c.dirs {
+		if isTerminal {
```

Comment:
IIRC in sigs.k8s.io/kind similar checks were problematic. Unless we can say that this works for sure everywhere, maybe we should just remove the special text formatting.

Reply:
maybe this one https://github.com/kubernetes-sigs/kind/pull/1479 or https://github.com/kubernetes-sigs/kind/pull/1924

will remove this special text formatting